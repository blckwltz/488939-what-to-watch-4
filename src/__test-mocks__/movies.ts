import {Movie} from '../types/movie';

const featuredMovie: Movie = {
  id: 1,
  isFavorite: false,
  title: `Pulp Fiction`,
  genre: `Drama`,
  releaseDate: 1994,
  runTime: 130,
  cover: `img/the-grand-budapest-hotel-poster.jpg`,
  poster: `img/pulp-fiction.jpg`,
  backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
  backgroundColor: `#ffffff`,
  videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: {
    score: 8.9,
    count: 578,
  },
  description: `The lives of two mob hitmen, a boxer, a gangster and his wife, and a pair of diner bandits
      intertwine in four tales of violence and redemption.`,
  director: `Quentin Tarantino`,
  cast: [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
};

const moviesList: Movie[] = [
  {
    id: 1,
    isFavorite: false,
    title: `Macbeth`,
    genre: `Drama`,
    releaseDate: 2015,
    runTime: 120,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    poster: `img/macbeth.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundColor: `#ffffff`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: {
      score: 6.6,
      count: 120,
    },
    description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will
      become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king
      and takes the throne for himself.`,
    director: `Justin Kurzel`,
    cast: [`Michael Fassbender`, `Marion Cotillard`, `Jack Madigan`],
  },
  {
    id: 2,
    isFavorite: false,
    title: `We need to talk about Kevin`,
    genre: `Drama`,
    releaseDate: 2011,
    runTime: 100,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    poster: `img/we-need-to-talk-about-kevin.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundColor: `#ffffff`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: {
      score: 7.5,
      count: 134,
    },
    description: `Kevin's mother struggles to love her strange child, despite the increasingly dangerous things
      he says and does as he grows up. But Kevin is just getting started, and his final act will be beyond
      anything anyone imagined.`,
    director: `Lynne Ramsay`,
    cast: [`Tilda Swinton`, `John C. Reilly`, `Ezra Miller`],
  },
  {
    id: 3,
    isFavorite: false,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `Adventure`,
    releaseDate: 2018,
    runTime: 185,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundColor: `#ffffff`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: {
      score: 6.6,
      count: 300,
    },
    description: `The second installment of the "Fantastic Beasts" series featuring the adventures
      of Magizoologist Newt Scamander.`,
    director: `David Yates`,
    cast: [`Eddie Redmayne`, `Katherine Waterston`, `Dan Fogler`],
  },
  {
    id: 4,
    isFavorite: false,
    title: `Shutter Island`,
    genre: `Mystery`,
    releaseDate: 2010,
    runTime: 154,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    poster: `img/shutter-island.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundColor: `#ffffff`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: {
      score: 8.1,
      count: 257,
    },
    description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer who escaped from a hospital
      for the criminally insane.`,
    director: `Martin Scorsese`,
    cast: [`Leonardo DiCaprio`, `Emily Mortimer`, `Mark Ruffalo`],
  },
  {
    id: 5,
    isFavorite: false,
    title: `Snatch`,
    genre: `Comedy`,
    releaseDate: 2000,
    runTime: 107,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    poster: `img/snatch.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundColor: `#ffffff`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: {
      score: 8.3,
      count: 300,
    },
    description: `Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers
      and supposedly Jewish jewelers fight to track down a priceless stolen diamond.`,
    director: `Guy Ritchie`,
    cast: [`Jason Statham`, `Brad Pitt`, `Benicio Del Toro`],
  },
  {
    id: 6,
    isFavorite: false,
    title: `Pulp Fiction`,
    genre: `Drama`,
    releaseDate: 1994,
    runTime: 130,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    poster: `img/pulp-fiction.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundColor: `#ffffff`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: {
      score: 8.9,
      count: 578,
    },
    description: `The lives of two mob hitmen, a boxer, a gangster and his wife, and a pair of diner bandits
      intertwine in four tales of violence and redemption.`,
    director: `Quentin Tarantino`,
    cast: [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
  },
  {
    id: 7,
    isFavorite: false,
    title: `Revenant`,
    genre: `Adventure`,
    releaseDate: 2015,
    runTime: 120,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    poster: `img/revenant.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundColor: `#ffffff`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: {
      score: 8.0,
      count: 1001,
    },
    description: `A frontiersman on a fur trading expedition in the 1820s fights for survival after being
      mauled by a bear and left for dead by members of his own hunting team.`,
    director: `Alejandro G. Iñárritu`,
    cast: [`Leonardo DiCaprio`, `Tom Hardy`, `Will Poulter`],
  },
  {
    id: 8,
    isFavorite: false,
    title: `Bohemian Rhapsody`,
    genre: `Biography`,
    releaseDate: 2018,
    runTime: 170,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    poster: `img/bohemian-rhapsody.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundColor: `#ffffff`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: {
      score: 8.0,
      count: 240,
    },
    description: `The story of the legendary British rock band Queen and lead singer Freddie Mercury,
      leading up to their famous performance at Live Aid (1985).`,
    director: `Bryan Singer`,
    cast: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
  },
];

export {featuredMovie, moviesList};
