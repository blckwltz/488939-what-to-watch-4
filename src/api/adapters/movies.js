const createMovie = (movie) => ({
  id: movie.id,
  isFavorite: movie[`is_favorite`],
  title: movie.name,
  genre: movie.genre,
  releaseDate: movie.released,
  runTime: movie[`run_time`],
  cover: movie[`poster_image`],
  poster: movie[`preview_image`],
  backgroundImage: movie[`background_image`],
  backgroundColor: movie[`background_color`],
  videoSrc: movie[`video_link`],
  previewSrc: movie[`preview_video_link`],
  rating: {
    score: movie.rating,
    count: movie[`scores_count`],
  },
  description: movie.description,
  director: movie.director,
  cast: movie.starring,
});

const createMoviesList = (movies) => {
  return movies.map(createMovie);
};

export {createMovie, createMoviesList};
