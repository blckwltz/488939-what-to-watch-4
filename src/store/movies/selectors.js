import {createSelector} from "reselect";
import {FilterSettings} from '../../utils/const.js';
import NameSpace from '../name-space.js';

const NAME_SPACE = NameSpace.MOVIES;

const getFeaturedMovie = (state) => {
  return state[NAME_SPACE].featuredMovie;
};

const getMoviesList = (state) => {
  return state[NAME_SPACE].moviesList;
};

const getStatus = (state) => {
  return state[NAME_SPACE].status;
};

const getShownMoviesAmount = (state) => {
  return state[NAME_SPACE].shownMoviesAmount;
};

const getActiveGenre = (state) => {
  return state[NAME_SPACE].activeGenre;
};

const getFilteredList = createSelector(
    getMoviesList,
    getActiveGenre,
    (list, genre) => {
      return genre === FilterSettings.INITIAL_VALUE ? list : list.filter((item) => {
        return item.genre === genre;
      });
    }
);

export {getFeaturedMovie, getMoviesList, getStatus, getActiveGenre, getShownMoviesAmount, getFilteredList};
