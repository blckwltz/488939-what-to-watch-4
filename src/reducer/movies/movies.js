import {FilterSettings, MAX_MOVIES_AMOUNT} from '../../utils/const.js';
import {extend} from '../../utils/utils.js';
import {createMovie, createMoviesList} from '../../api/adapters/movies.js';

const initialState = {
  featuredMovie: {},
  moviesList: [],
  activeGenre: FilterSettings.INITIAL_VALUE,
  shownMoviesAmount: MAX_MOVIES_AMOUNT,
};

const ActionType = {
  LOAD_FEATURED_MOVIE: `LOAD_FEATURED_MOVIE`,
  LOAD_MOVIES: `LOAD_MOVIES`,
  SET_ACTIVE_GENRE: `SET_ACTIVE_GENRE`,
  SET_SHOWN_AMOUNT: `SET_SHOWN_AMOUNT`,
};

const ActionCreator = {
  loadFeaturedMovie: (movie) => {
    return {
      type: ActionType.LOAD_FEATURED_MOVIE,
      payload: movie,
    };
  },
  loadMovies: (moviesList) => {
    return {
      type: ActionType.LOAD_MOVIES,
      payload: moviesList,
    };
  },
  getFilteredList: (genre) => ({
    type: ActionType.SET_ACTIVE_GENRE,
    payload: genre,
  }),
  setShownAmount: () => ({
    type: ActionType.SET_SHOWN_AMOUNT,
    payload: MAX_MOVIES_AMOUNT,
  }),
};

const Operation = {
  loadFeaturedMovie: () => (dispatch, getState, api) => {
    return api.get(`/films/promo`)
      .then((response) => {
        dispatch(ActionCreator.loadFeaturedMovie(createMovie(response.data)));
      });
  },
  loadMovies: () => (dispatch, getState, api) => {
    return api.get(`/films`)
      .then((response) => {
        dispatch(ActionCreator.loadMovies(createMoviesList(response.data)));
      });
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.LOAD_FEATURED_MOVIE:
      return extend(state, {
        featuredMovie: action.payload,
      });
    case ActionType.LOAD_MOVIES:
      return extend(state, {
        moviesList: action.payload,
      });
    case ActionType.SET_ACTIVE_GENRE:
      return extend(state, {
        activeGenre: action.payload,
        shownMoviesAmount: MAX_MOVIES_AMOUNT,
      });
    case ActionType.SET_SHOWN_AMOUNT:
      return extend(state, {
        shownMoviesAmount: state.shownMoviesAmount + action.payload,
      });
  }

  return state;
};


export {reducer, Operation, ActionType, ActionCreator};
