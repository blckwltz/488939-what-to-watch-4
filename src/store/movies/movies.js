import {URL, Status, FilterSettings, MAX_MOVIES_AMOUNT} from '../../utils/const.js';
import {extend} from '../../utils/utils.js';
import {createMovie, createMoviesList} from '../../api/adapters/movies.js';

const initialState = {
  featuredMovie: {},
  moviesList: [],
  activeGenre: FilterSettings.INITIAL_VALUE,
  shownMoviesAmount: MAX_MOVIES_AMOUNT,
  status: Status.OK,
};

const ActionType = {
  LOAD_FEATURED_MOVIE: `LOAD_FEATURED_MOVIE`,
  LOAD_MOVIES: `LOAD_MOVIES`,
  UPDATE_STATUS: `UPDATE_STATUS`,
  SET_ACTIVE_GENRE: `SET_ACTIVE_GENRE`,
  SET_SHOWN_AMOUNT: `SET_SHOWN_AMOUNT`,
  UPDATE_MOVIE_STATUS: `UPDATE_MOVIE_STATUS`,
};

const ActionCreator = {
  loadFeaturedMovie: (movie) => {
    return {
      type: ActionType.LOAD_FEATURED_MOVIE,
      payload: movie,
    };
  },
  loadMovies: (moviesList) => {
    return {
      type: ActionType.LOAD_MOVIES,
      payload: moviesList,
    };
  },
  updateStatus: (status) => {
    return {
      type: ActionType.UPDATE_STATUS,
      payload: status,
    };
  },
  getFilteredList: (genre) => ({
    type: ActionType.SET_ACTIVE_GENRE,
    payload: genre,
  }),
  setShownAmount: () => ({
    type: ActionType.SET_SHOWN_AMOUNT,
    payload: MAX_MOVIES_AMOUNT,
  }),
  updateMovieStatus: (movie) => ({
    type: ActionType.UPDATE_MOVIE_STATUS,
    payload: movie,
  }),
};

const Operation = {
  loadFeaturedMovie: () => (dispatch, getState, api) => {
    return api.get(URL.FEATURED)
      .then((response) => {
        dispatch(ActionCreator.loadFeaturedMovie(createMovie(response.data)));
      })
      .catch(() => {
        dispatch(ActionCreator.updateStatus(Status.BAD_REQUEST));
      });
  },
  loadMovies: () => (dispatch, getState, api) => {
    return api.get(URL.MOVIES)
      .then((response) => {
        dispatch(ActionCreator.loadMovies(createMoviesList(response.data)));
      })
      .catch(() => {
        dispatch(ActionCreator.updateStatus(Status.BAD_REQUEST));
      });
  },
  updateMovieStatus: (id, status) => (dispatch, getState, api) => {
    return api.post(`${URL.FAVORITE}/${id}/${status}`)
      .then((response) => {
        dispatch(ActionCreator.updateMovieStatus(createMovie(response.data)));
      });
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.LOAD_FEATURED_MOVIE:
      return extend(state, {
        featuredMovie: action.payload,
      });
    case ActionType.LOAD_MOVIES:
      return extend(state, {
        moviesList: action.payload,
      });
    case ActionType.UPDATE_STATUS:
      return extend(state, {
        status: action.payload,
      });
    case ActionType.SET_ACTIVE_GENRE:
      return extend(state, {
        activeGenre: action.payload,
        shownMoviesAmount: MAX_MOVIES_AMOUNT,
      });
    case ActionType.SET_SHOWN_AMOUNT:
      return extend(state, {
        shownMoviesAmount: state.shownMoviesAmount + action.payload,
      });
    case ActionType.UPDATE_MOVIE_STATUS:
      const updateMovieStatus = () => {
        const {isFavorite} = action.payload;
        const featuredMovie = state.featuredMovie;
        const moviesList = state.moviesList;
        const targetMovie = moviesList.find((item) => {
          return item.id === action.payload.id;
        });

        if (featuredMovie.id === targetMovie.id) {
          featuredMovie.isFavorite = isFavorite;
        }

        targetMovie.isFavorite = isFavorite;

        return {
          featuredMovie,
          moviesList,
        };
      };

      return extend(state, updateMovieStatus());
  }

  return state;
};


export {reducer, Operation, ActionType, ActionCreator};
