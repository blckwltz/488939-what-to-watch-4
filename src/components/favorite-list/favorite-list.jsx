import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import {Operation as UserOperation} from '../../store/user/user.js';
import {getFavoriteList} from '../../store/user/selectors.js';
import {AppRoute} from '../../routing/route.js';
import UserBlock from '../user-block/user-block.jsx';
import MoviesList from '../movies-list/movies-list.jsx';

class FavoriteList extends PureComponent {
  constructor(props) {
    super(props);

    this._handleFavoriteListLoad = this._handleFavoriteListLoad.bind(this);
  }

  componentDidMount() {
    this._handleFavoriteListLoad();
  }

  _handleFavoriteListLoad() {
    const {onLoad} = this.props;

    onLoad();
  }

  render() {
    const {favoriteList} = this.props;

    if (!favoriteList) {
      return null;
    }

    return <div className="user-page">
      <header className="page-header user-page__head">
        <div className="logo">
          <Link to={AppRoute.ROOT} className="logo__link">
            <span className="logo__letter logo__letter--1">W</span>
            <span className="logo__letter logo__letter--2">T</span>
            <span className="logo__letter logo__letter--3">W</span>
          </Link>
        </div>

        <h1 className="page-title user-page__title">My list</h1>

        <UserBlock/>
      </header>

      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>

        <MoviesList movies={favoriteList}/>
      </section>

      <footer className="page-footer">
        <div className="logo">
          <Link to={AppRoute.ROOT} className="logo__link logo__link--light">
            <span className="logo__letter logo__letter--1">W</span>
            <span className="logo__letter logo__letter--2">T</span>
            <span className="logo__letter logo__letter--3">W</span>
          </Link>
        </div>

        <div className="copyright">
          <p>Â© 2019 What to watch Ltd.</p>
        </div>
      </footer>
    </div>;
  }
}

FavoriteList.propTypes = {
  favoriteList: PropTypes.array,
  onLoad: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  favoriteList: getFavoriteList(state),
});

const mapDispatchToProps = (dispatch) => ({
  onLoad() {
    dispatch(UserOperation.loadFavoriteList());
  }
});

export {FavoriteList};
export default connect(mapStateToProps, mapDispatchToProps)(FavoriteList);
