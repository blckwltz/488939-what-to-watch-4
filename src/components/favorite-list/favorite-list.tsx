import React, {PureComponent} from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import {Movie} from '../../types/movie';
import {Operation as UserOperation} from '../../store/user/user';
import {getFavoriteList} from '../../store/user/selectors';
import {AppRoute} from '../../routing/route';
import UserBlock from '../user-block/user-block';
import MoviesList from '../movies-list/movies-list';

interface Props {
  favoriteList: Movie[];
  onLoad: () => void;
}

class FavoriteList extends PureComponent<Props> {
  constructor(props) {
    super(props);

    this._handleFavoriteListLoad = this._handleFavoriteListLoad.bind(this);
  }

  componentDidMount() {
    this._handleFavoriteListLoad();
  }

  _handleFavoriteListLoad() {
    const {onLoad} = this.props;

    onLoad();
  }

  render() {
    const {favoriteList} = this.props;

    if (!favoriteList) {
      return null;
    }

    return <div className="user-page">
      <header className="page-header user-page__head">
        <div className="logo">
          <Link to={AppRoute.ROOT} className="logo__link">
            <span className="logo__letter logo__letter--1">W</span>
            <span className="logo__letter logo__letter--2">T</span>
            <span className="logo__letter logo__letter--3">W</span>
          </Link>
        </div>

        <h1 className="page-title user-page__title">My list</h1>

        <UserBlock/>
      </header>

      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>

        <MoviesList movies={favoriteList}/>
      </section>

      <footer className="page-footer">
        <div className="logo">
          <Link to={AppRoute.ROOT} className="logo__link logo__link--light">
            <span className="logo__letter logo__letter--1">W</span>
            <span className="logo__letter logo__letter--2">T</span>
            <span className="logo__letter logo__letter--3">W</span>
          </Link>
        </div>

        <div className="copyright">
          <p>Â© 2020 What to watch Ltd.</p>
        </div>
      </footer>
    </div>;
  }
}

const mapStateToProps = (state) => ({
  favoriteList: getFavoriteList(state),
});

const mapDispatchToProps = (dispatch) => ({
  onLoad() {
    dispatch(UserOperation.loadFavoriteList());
  }
});

export {FavoriteList};
export default connect(mapStateToProps, mapDispatchToProps)(FavoriteList);
