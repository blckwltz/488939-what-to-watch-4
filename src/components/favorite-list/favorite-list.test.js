import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {MemoryRouter} from 'react-router-dom';
import {moviesList} from '../../__test-mocks__/movies.js';
import NameSpace from '../../store/name-space.js';
import {FavoriteList} from './favorite-list';

const mockStore = configureStore([]);
const onLoad = () => {};

it(`Should render FavoriteList component correctly`, () => {
  const store = mockStore({
    [NameSpace.MOVIES]: {
      moviesList,
    },
    [NameSpace.USER]: {
      authorizationStatus: `AUTH`,
    }
  });
  const tree = renderer
    .create((
      <Provider store={store}>
        <MemoryRouter>
          <FavoriteList
            favoriteList={moviesList}
            onLoad={onLoad}
          />
        </MemoryRouter>
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
