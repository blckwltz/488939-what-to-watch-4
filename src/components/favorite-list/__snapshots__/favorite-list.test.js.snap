// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render FavoriteList component correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="/img/avatar.jpg"
            width="63"
          />
        </div>
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/1"
            onClick={[Function]}
          >
            <img
              alt="Macbeth"
              height="175"
              src="img/macbeth.jpg"
              width="280"
            />
          </a>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
          >
            Macbeth
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/2"
            onClick={[Function]}
          >
            <img
              alt="We need to talk about Kevin"
              height="175"
              src="img/we-need-to-talk-about-kevin.jpg"
              width="280"
            />
          </a>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/2"
            onClick={[Function]}
          >
            We need to talk about Kevin
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/3"
            onClick={[Function]}
          >
            <img
              alt="Fantastic Beasts: The Crimes of Grindelwald"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </a>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/3"
            onClick={[Function]}
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/4"
            onClick={[Function]}
          >
            <img
              alt="Shutter Island"
              height="175"
              src="img/shutter-island.jpg"
              width="280"
            />
          </a>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/4"
            onClick={[Function]}
          >
            Shutter Island
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/5"
            onClick={[Function]}
          >
            <img
              alt="Snatch"
              height="175"
              src="img/snatch.jpg"
              width="280"
            />
          </a>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/5"
            onClick={[Function]}
          >
            Snatch
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/6"
            onClick={[Function]}
          >
            <img
              alt="Pulp Fiction"
              height="175"
              src="img/pulp-fiction.jpg"
              width="280"
            />
          </a>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/6"
            onClick={[Function]}
          >
            Pulp Fiction
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/7"
            onClick={[Function]}
          >
            <img
              alt="Revenant"
              height="175"
              src="img/revenant.jpg"
              width="280"
            />
          </a>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/7"
            onClick={[Function]}
          >
            Revenant
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/8"
            onClick={[Function]}
          >
            <img
              alt="Bohemian Rhapsody"
              height="175"
              src="img/bohemian-rhapsody.jpg"
              width="280"
            />
          </a>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/8"
            onClick={[Function]}
          >
            Bohemian Rhapsody
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2020 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
